### PR TITLE
Adding quotes around  in password_hash line to be compliant with puppet 4.x

### PR DESCRIPTION
**Pull request by jacobmw**

### CODE REVIEW

File: manifests/postgres.pp [hunk 1]
```diff
@@ -173,7 +173,7 @@
   # define user used by Barman to connect into PostgreSQL database(s)
   postgresql::server::role { $barman_dbuser:
     login         => true,
-    password_hash => postgresql_password($barman_dbuser, $real_password),
+    password_hash => postgresql_password($barman_dbuser, inline_template('<%= @real_password.to_s %>')),
```
Comment (igalic) [inline manifests/postgres.pp]: this seems like unnecessary force…

Comment (jacobmw) [inline manifests/postgres.pp]: This was the best suggestion that I got from the Puppet Users group.  It gets the desired result and puppet-lint doesn't throw any errors.  Do you have another idea to help get this module be compatible with puppet > 4.x?

Comment (sathieu) [inline manifests/postgres.pp]: Was it the problem you are trying to fix here?

Comment (jacobmw) [inline manifests/postgres.pp]: The 'problem' just has to do with changes that were made in puppet >=4.0.  Not actually trying to make anything different happen, just making the syntax work with newer version of puppet and puppet-lint.

Comment (sathieu) [inline manifests/postgres.pp]: Your patch suggests that $password (and $real_password) is not a string. How?

**Comment by jacobmw** [comment]
Here is the error that is generated using the existing code on a barman client host:
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Error while evaluating a Function Call, String can't be coerced into Fixnum at /etc/puppetlabs/code/environments/barman/modules/barman/manifests/postgres.pp:176:22

**Comment by sathieu** [comment]
@jacobmw Then use: https://github.com/puppetlabs/puppetlabs-postgresql/pull/774, and drop this commit.

**Comment by fcanovai** [comment]
@jacobmw I'd avoid the inline template and use a random string instead in case of an empty password (see 348eb3f837638fb26d554a1b9d84f926ca9d2c85). I believe the responsibility of accepting integers is on the postgres module, as per @sathieu's patch.

**Pull request closed without merging**